crates/crypto/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+use calimero_primitives::identity::{PrivateKey, PublicKey};
+use curve25519_dalek::Scalar;
+use ed25519_dalek::SecretKey;
+use ring::aead;
+use serde::{Deserialize, Serialize};
+
+#[derive(Copy, Clone, Debug, Serialize, Deserialize)]
+pub struct SharedKey {
+    key: SecretKey,
+}
+
+#[derive(Debug)]
+pub struct Record {
+    pub token: Vec<u8>,
+    pub nonce: [u8; aead::NONCE_LEN],
+}
+
+impl SharedKey {
+    pub fn new(sk: &PrivateKey, pk: &PublicKey) -> Self {
+        SharedKey {
+            key: (Scalar::from_bytes_mod_order(**sk)
```

Comment:
do the tests pass? I'd expect `Scalar::from_bytes_mod_order(sha512(bytes)[..32])`, so it shouldn't work.

any reason we're not using [`SigningKey::from_bytes`](https://docs.rs/ed25519-dalek/latest/ed25519_dalek/struct.SigningKey.html#method.from_bytes)?